src/filesystem.cpp

```diff
@@ -261,33 +261,58 @@ bool is_filename_case_correct(const std::string& /*fname*/, const boost::iostrea
 namespace filesystem
 {
 
+/**
+ * Like a .gitignore file, it's useful to exclude backup files created by editors and similar when
+ * uploading add-ons. Loading those files is also blocked, so that add-ons which work locally are
+ * less likely to be broken by missing files after uploading.
+ *
+ * The only security part of this is blocking local access to .pbl files, as they can contain
+ * passwords yet are in the area that scripts can access.
+ */
 const blacklist_pattern_list default_blacklist{
+	/* The first block is exact filenames */
 	{
-		/* Blacklist dot-files/dirs, which are hidden files in UNIX platforms */
-		".+",
-		"#*#",
-		"*~",
-		"*-bak",
-		"*.swp",
-		"*.pbl",
-		"*.ign",
+		/* If an add-on is downloaded then this will be replaced with an autogenerated file. */
 		"_info.cfg",
-		"*.exe",
-		"*.bat",
-		"*.cmd",
-		"*.com",
-		"*.scr",
-		"*.sh",
-		"*.js",
-		"*.vbs",
-		"*.o",
-		"*.ini",
 		/* Remove junk created by certain file manager ;) */
 		"Thumbs.db",
+	},
+	/* The second block is file extensions (checked simply with ends_with for speed) */
+	{
+		/* _server.pbl may contain passwords, so these shouldn't be accessible to scripts */
+		".pbl",
+
+		/* Backups created by editors or wmllint */
+		"~",
+		"-bak",
+		".swp", // Vim's swap files? Modern versions put a dot at the start of the filename too
+		".ign",
+		".orig", // Git's leftovers from merge conflicts
+
+		/* These aren't expected in the add-ons dir, and might contain trojans. Client-side security
+		 * can't replace server-side security, but we might as well prevent accidental uploads. */
+		".exe",
+		".bat",
+		".cmd",
+		".com",
+		".scr",
+		".sh",
```

Comment:
Some of my add-ons contain shell scripts in them (usually as a wrapper for updating translations with `msgmerge`), just FYI

Reply:
That sounds like something that doesn't need to be uploaded though.

Reply:
I guess that's true...